spec/classes/puppet_server_puppetserver_spec.rb

```diff
@@ -276,10 +277,33 @@
         end
       end
 
-      describe 'server_trusted_agents' do
-        context 'when set' do
+      describe 'puppetlabs v3 catalog from agents rule' do
+        context 'when server_trusted_agents set' do
           let(:params) { super().merge(server_puppetserver_trusted_agents: ['jenkins', 'octocatalog-diff']) }
-          it { should contain_file(auth_conf).with_content(/^            allow: \["jenkins", "octocatalog-diff", "\$1"\]$/) }
+          it do
+            content = catalogue.resource('file', auth_conf).send(:parameters)[:content]
+            rules = Hocon.parse(content)['authorization']['rules']
+            rule = rules.find {|rule| rule['name'] == 'puppetlabs v3 catalog from agents' }
```

Comment:
Looks like you can extract this to a `let(:rule)` block and reuse it between the 3 tests

Reply:
and I guess `rules` and `content` too.